api/src/main/java/edu/cornell/mannlib/vitro/webapp/rdfservice/impl/RDFServiceImpl.java

```diff
@@ -32,109 +42,115 @@
-import edu.cornell.mannlib.vitro.webapp.rdfservice.ModelChange.Operation;
-import edu.cornell.mannlib.vitro.webapp.rdfservice.RDFService;
-import edu.cornell.mannlib.vitro.webapp.rdfservice.RDFServiceException;
-import edu.cornell.mannlib.vitro.webapp.rdfservice.ResultSetConsumer;
-import edu.cornell.mannlib.vitro.webapp.utils.logging.ToString;
-
 public abstract class RDFServiceImpl implements RDFService {
 
-	private static final Log log = LogFactory.getLog(RDFServiceImpl.class);
-	
-	protected String defaultWriteGraphURI;
-	protected List<ChangeListener> registeredListeners = new CopyOnWriteArrayList<ChangeListener>();
-	protected List<ModelChangedListener> registeredJenaListeners = new CopyOnWriteArrayList<ModelChangedListener>();
+    private static final Log log = LogFactory.getLog(RDFServiceImpl.class);
 
-	@Override
-	public void newIndividual(String individualURI,
-			                  String individualTypeURI) throws RDFServiceException {
+    protected String defaultWriteGraphURI;
+    protected List<ChangeListener> registeredListeners = new CopyOnWriteArrayList<ChangeListener>();
+    protected List<ModelChangedListener> registeredJenaListeners = new CopyOnWriteArrayList<ModelChangedListener>();
+    protected final List<String> graphURIs = new CopyOnWriteArrayList<String>();
+    protected volatile boolean rebuildGraphURICache = true;
+    protected volatile boolean isRebuildGraphURICacheRunning = false;
 
-       newIndividual(individualURI, individualTypeURI, defaultWriteGraphURI);
-	}
+    @Override
+    public void newIndividual(String individualURI, String individualTypeURI) throws RDFServiceException {
+
+        newIndividual(individualURI, individualTypeURI, defaultWriteGraphURI);
+    }
 
     @Override
-    public void newIndividual(String individualURI,
-                              String individualTypeURI,
-                              String graphURI) throws RDFServiceException {
-
-       StringBuilder containsQuery = new StringBuilder("ASK { \n");
-       if (graphURI != null) {
-           containsQuery.append("  GRAPH <").append(graphURI).append("> { ");
-       }
-       containsQuery.append("<");
-       containsQuery.append(individualURI);
-       containsQuery.append("> ");
-       containsQuery.append("?p ?o");
-       if (graphURI != null) {
-           containsQuery.append(" } \n");
-       }
-       containsQuery.append("\n}");
-
-       if (sparqlAskQuery(containsQuery.toString())) {
+    public void newIndividual(String individualURI, String individualTypeURI, String graphURI)
+            throws RDFServiceException {
+
+        StringBuilder containsQuery = new StringBuilder("ASK { \n");
+        if (graphURI != null) {
+            containsQuery.append("  GRAPH <").append(graphURI).append("> { ");
+        }
+        containsQuery.append("<");
+        containsQuery.append(individualURI);
+        containsQuery.append("> ");
+        containsQuery.append("?p ?o");
```

Comment:
Can you chain these append statements in a single line? I think that will make it more concise and readable.